apps/niftest/niftest.cpp

```diff
@@ -128,14 +129,24 @@ std::vector<std::string> parseOptions (int argc, char** argv)
 
 int main(int argc, char **argv)
 {
-    std::vector<std::string> files = parseOptions (argc, argv);
+    std::vector<std::string> files;
+    try
+    {
+        files = parseOptions (argc, argv);
+    }
+    catch( boost::exception &e )
+    {
+        std::cout << "ERROR parsing arguments: " << boost::diagnostic_information(e) << std::endl;
+        exit(1);
```

Comment:
Don't use exit in C++ code. It is not guaranteed to do its job cleanly (some cleanup may be left undone). Doesn't look like that is the problem here, but  it is still kinda dirty. I suggest to use return 1 instead.

Also, is there actually a need to pull in boost exceptions? I would suspect that a std::exception should do the job just fine.

Reply:
1. NIFtest already uses an `exit(1)` in different places. If we do not like it, we need to refactor it.
2. Boost exceptions are not derived from std::exception, so std::exception catch more likely will not work, and we will need to catch them separately.
We already use a similar approach in the BSATool::parseOptions().

Reply:
niftest is a CLI tool and not a library.